### PR TITLE
Fix ordering for reinscriptions and show all reinscriptions for sat

### PR DESCRIPTION
**Pull request by veryordinally**
This PR fixes a bug in the assignment of reinscription sequence numbers and allows to correctly order reinscriptions in chronological order. It also implements showing all inscriptions for a given sat (and not just a random one) in the correct sequence.

### CODE REVIEW

**Comment by veryordinally** [comment]
See https://github.com/raphjaph/ord/commit/98340591c2981bad6d3215b185e1c9bc7c3e0f66 for where bug was introduced

File: src/subcommand/wallet/inscriptions.rs [hunk 1]
```diff
@@ -11,6 +11,8 @@ pub(crate) fn run(options: Options) -> Result {
   let index = Index::open(&options)?;
   index.update()?;
 
+  // BACKLOG - change the logic here to only get inscriptions for outputs actually in the wallet
```
Comment (raphjaph) [inline src/subcommand/wallet/inscriptions.rs:14]: Make issue

File: templates/sat.html [hunk 2]
```diff
@@ -11,8 +11,13 @@ <h1>Sat {{ self.sat.n() }}</h1>
   <dt>offset</dt><dd>{{ self.sat.third() }}</dd>
   <dt>rarity</dt><dd><span class={{self.sat.rarity()}}>{{ self.sat.rarity() }}</span></dd>
   <dt>timestamp</dt><dd><time>{{self.blocktime.timestamp()}}</time>{{self.blocktime.suffix()}}</dd>
-%% if let Some((inscription)) = &self.inscription {
-  <dt>inscription</dt><dd class=thumbnails>{{ Iframe::thumbnail(*inscription) }}</dd>
+%% if !self.inscriptions.is_empty() {
```
Comment (raphjaph) [inline templates/sat.html:14]: Did you visually confirm that this looks good?

File: src/index.rs [hunk 3]
```diff
@@ -536,17 +536,30 @@ impl Index {
...
+          .ok()
+          .map(|inscription_id| InscriptionId::load(*inscription_id.value()))
+      })
+      .collect::<Vec<InscriptionId>>();
+
+    if ids.len() > 1 {
+      let re_id_to_seq_num = rtx.open_table(REINSCRIPTION_ID_TO_SEQUENCE_NUMBER)?;
+      ids.sort_by_key(
+        |inscription_id| match re_id_to_seq_num.get(&inscription_id.store()) {
+          Ok(Some(num)) => num.value() + 1,
```
Comment (raphjaph) [inline src/index.rs:556]: Why the +1?

File: src/index.rs [hunk 4]
```diff
@@ -989,7 +995,7 @@ impl Index {
 
     result.sort_by_key(|(_satpoint, inscription_id)| {
       match re_id_to_seq_num.get(&inscription_id.store()) {
-        Ok(Some(num)) => num.value(),
+        Ok(Some(num)) => num.value() + 1,
```
Comment (raphjaph) [inline src/index.rs:998]: Why +1?

File: templates/sat.html (hunk 2 above)
Comment (veryordinally) [reply templates/sat.html:14]: Yes. It's the same layout we use for showing all inscriptions in an output.

File: src/index.rs (hunk 3 above)
Comment (veryordinally) [reply src/index.rs:556]: We want the first inscription to have the smallest value in the ordering for this comparison, so we use 0 when the inscription is not in the table. That means first reinscription should get 1 to be greater than 0. Sequence numbers in the table start with 0 (that was part of the bug introduced ) and I wanted to fix this in a way that does not require an index rebuild.

File: src/index.rs (hunk 4 above)
Comment (veryordinally) [reply src/index.rs:998]: See previous comment

File: src/subcommand/wallet/inscriptions.rs (hunk 1 above)
Comment (veryordinally) [reply src/subcommand/wallet/inscriptions.rs:14]: Done: #2282

**Review by raphjaph — APPROVED** [review]
Added some comments to remove the +1 the next time we refactor the database.

**Pull request merged**